src/core/qgslegendrenderer.cpp

```diff
@@ -928,7 +928,10 @@ QgsLegendRenderer::LegendComponent QgsLegendRenderer::drawSymbolItem( QgsLayerTr
   ctx.maxSiblingSymbolWidth = maxSiblingSymbolWidth;
 
   if ( const QgsSymbolLegendNode *symbolNode = dynamic_cast< const QgsSymbolLegendNode * >( symbolItem ) )
+  {
+    context.expressionContext().appendScope( symbolNode->createSymbolScope() );
```

Comment:
We'll need to pop this scope after drawing the symbol item, or the expression context risks becoming huge when there's many symbols present in the legend.

Reply:
Good catch!

Fixed in 89dfb017f20295e2a739a07c92f2a2e05967a468

Thanks for the review